ios/engine/KMEI/KeymanEngine/resources/Keyman.bundle/Contents/Resources/ios-host.js

```diff
@@ -264,30 +261,34 @@ function doResetContext() {
 }
 
 function setCursorRange(pos, length) {
+    var context = keyman.context;
+
     //console.log('setCursorRange('+pos+', '+length+')');
-    var ta = document.getElementById('ta');
-    var resetContext = (ta.selectionStart != pos || ta.selectionEnd != pos + length);
-    ta.selectionStart = ta._KeymanWebSelectionStart = pos;
-    ta.selectionEnd = ta._KeymanWebSelectionEnd = pos + length;
-    if(resetContext) {
-        //console.log('  setCursorRange: resetting context');
-        doResetContext();
+
+    var start = pos;
+    var end = pos + length;
+
+    if(context.selStart != start || context.selEnd != end) {
+      keyman.context.setSelection(start, end);
+      keyman.resetContext();
     }
-    return ta.selectionEnd;
 }
 
 function setKeymanVal(text) {
     //console.log('setKeymanVal('+JSON.stringify(text)+')');
-    if(undefined == text) text = '';
-    var ta = document.getElementById('ta');
 
-    var resetContext = ta.value != text;
-    ta.value = text;
+    if(text == undefined) {
+        text = '';
+    }
+
+    var resetContext = keyman.context.getText() != text;
+
     if(resetContext) {
```

Comment:
```suggestion
    if(keyman.context.getText() != text) {
```

Let's fix that unnecessary var.